docs/charts.md

```diff
@@ -763,14 +764,72 @@ parent chart.
 
 Also, global variables of parent charts take precedence over the global variables from subcharts.
 
+### Schema Files
+
+Sometimes, a chart writer might want to define a structure on their values.
+This can be done by defining a schema in the `values.schema.yaml` file. A
+schema is the yaml representation of a [JSON Schema](https://json-schema.org/).
+It might look something like this:
+
+```yaml
+title: Values
+type: object
+properties:
+    name:
+        description: Service name
+        type: string
+    protocol:
+        type: string
+    port:
+        description: Port
+        type: integer
+        minimum: 0
+    image:
+        description: Container Image
+        type: object
+        properties:
+            repo:
+                type: string
+            tag:
+                type: string
+required:
+    - protocol
+    - port
+```
+
+This schema will be applied to the values to validate it. An example of a
```

Comment:
Would you mind calling out when the values are validated? `helm template`, `helm install`, `helm upgrade`, `helm lint`, etc.

Reply:
done